src/rpc/rawtransaction.cpp

```diff
@@ -1739,6 +1741,334 @@ UniValue converttopsbt(const JSONRPCRequest& request)
+            "      \"is_final\" : true|false     (boolean) Whether the input is finalized\n"
+            "      \"missing\" : {               (json object, optional) Things that are missing that are required to complete this input\n"
+            "        \"pubkeys\" : [             (array)\n"
+            "          \"keyid\"                 (string) Public key ID, hash160 of the public key, of a public key whose BIP 32 derivation path is missing\n"
+            "        ]\n"
+            "        \"signatures\" : [          (array)\n"
+            "          \"keyid\"                 (string) Public key ID, hash160 of the public key, of a public key whose signature is missing\n"
+            "        ]\n"
+            "        \"redeemscript\" : \"hash\"   (string) Hash160 of the redeemScript that is missing\n"
+            "        \"witnessscript\" : \"hash\"  (string) SHA256 of the witnessScript that is missing\n"
+            "      }\n"
+            "      \"next\" : \"role\"           (string) Role of the next person that this input needs to go to\n"
+            "    }\n"
+            "    ,...\n"
+            "  ]\n"
+            "  \"estimated_vsize\" : vsize       (numeric) Estimated vsize of the final signed transaction\n"
+            "  \"estimated_feerate\" : feerate   (numeric, optional) Estimated feerate of the final signed transaction. Shown only if all UTXO slots in the PSBT have been filled.\n"
+            "  \"fee\" : fee                     (numeric, optional) The transaction fee paid. Shown only if all UTXO slots in the PSBT have been filled.\n"
+            "  \"next\" : \"role\"                 (string) Role of the next person that this psbt needs to go to\n"
+            "}\n"
+            "\nExamples:\n"
+            + HelpExampleCli("analyzepsbt", "\"psbt\"")
+            );
+
+    RPCTypeCheck(request.params, {UniValue::VSTR});
+
+    // Unserialize the transaction
+    PartiallySignedTransaction psbtx;
+    std::string error;
+    if (!DecodePSBT(psbtx, request.params[0].get_str(), error)) {
+        throw JSONRPCError(RPC_DESERIALIZATION_ERROR, strprintf("TX decode failed %s", error));
+    }
+
+    // Go through each input and build status
+    UniValue result(UniValue::VOBJ);
+    UniValue inputs_result(UniValue::VARR);
+    bool calc_fee = true;
+    bool all_final = true;
+    bool only_missing_sigs = false;
+    bool only_missing_final = false;
+    CAmount in_amt = 0;
+    for (unsigned int i = 0; i < psbtx.tx->vin.size(); ++i) {
+        PSBTInput& input = psbtx.inputs[i];
+        UniValue input_univ(UniValue::VOBJ);
+        UniValue missing(UniValue::VOBJ);
+
+        // Check for a UTXO
+        CTxOut utxo;
+        if (input.GetUTXO(utxo, psbtx.tx->vin[i].prevout.n)) {
+            in_amt += utxo.nValue;
+            input_univ.pushKV("has_utxo", true);
+        } else {
+            input_univ.pushKV("has_utxo", false);
+            input_univ.pushKV("is_final", false);
```

Comment:
Are there cases where this isn't overwritten by a later check? If so, you should set all_final = false here. (If not you should remove this.)

Reply:
Oh, I see, it's handled by calc_fee = false? The flow is a little hard to follow.

Reply:
I'm not quite sure you are asking.

Reply:
Yeah, I think I just have trouble following the state machine of this function, which is a little complicated, and so I was confused about whether all_final was actually getting set correctly in every case. Since nobody's relying on it yet, and I haven't found any actual issues, I think it should probably go in as-is, and I may propose some refactoring when I go to actually use it.